src/core/ext/filters/client_channel/http_connect_handshaker.cc

```diff
@@ -131,28 +130,29 @@ void HttpConnectHandshaker::HandshakeFailedLocked(grpc_error* error) {
 // Callback invoked when finished writing HTTP CONNECT request.
 void HttpConnectHandshaker::OnWriteDone(void* arg, grpc_error* error) {
   auto* handshaker = static_cast<HttpConnectHandshaker*>(arg);
-  gpr_mu_lock(&handshaker->mu_);
+  ReleasableMutexLock lock(&handshaker->mu_);
   if (error != GRPC_ERROR_NONE || handshaker->is_shutdown_) {
     // If the write failed or we're shutting down, clean up and invoke the
     // callback with the error.
     handshaker->HandshakeFailedLocked(GRPC_ERROR_REF(error));
-    gpr_mu_unlock(&handshaker->mu_);
+    lock.Unlock();
     handshaker->Unref();
   } else {
     // Otherwise, read the response.
     // The read callback inherits our ref to the handshaker.
-    grpc_endpoint_read(handshaker->args_->endpoint,
-                       handshaker->args_->read_buffer,
-                       &handshaker->response_read_closure_, /*urgent=*/true);
-    gpr_mu_unlock(&handshaker->mu_);
+    grpc_endpoint* ep = handshaker->args_->endpoint;
+    grpc_slice_buffer* read_buffer = handshaker->args_->read_buffer;
+    grpc_closure* closure = &handshaker->response_read_closure_;
+    lock.Unlock();
+    grpc_endpoint_read(ep, read_buffer, closure, /*urgent=*/true);
   }
 }
 
 // Callback invoked for reading HTTP CONNECT response.
 void HttpConnectHandshaker::OnReadDone(void* arg, grpc_error* error) {
   auto* handshaker = static_cast<HttpConnectHandshaker*>(arg);
 
```

Comment:
Not new in this PR, but please remove this blank line within a function.

Reply:
done